discordBot.js

```diff
@@ -1,57 +1,110 @@
-const Discord = require('discord.js');
+const {Client, MessageAttachment} = require('discord.js'),
+      http    = require('http');
 
 class DiscordBot {
-  constructor() {
-    const client = new Discord.Client();
-    this.client = client;
+  constructor(DISCORD_BOT_TOKEN) {
+    this.client = new Client();
+    this.token  = DISCORD_BOT_TOKEN;
   }
 
-  start() {
-    const token = process.env.DISCORD_BOT_TOKEN;
-    const client = this.client;
-    client.once('ready', () => {
-      console.log('Discord Ready!');
-    });
-
-    client.on('guildMemberAdd', member => {
-      const channel = member.guild.channels.cache.find(ch => ch.name === 'member-log');
-
-      if (!channel) return;
-      channel.send(`Welcome to the server, ${member}`);
-    });
-
-    client.on('message', async message => {
-      const PREFIX = '*';
-      if (message.content.startsWith(PREFIX)) {
-        const input = message.content.slice(PREFIX.length).trim().split(' ');
-        const command = input.shift();
-        const commandArgs = input.join(' ');
-
-        if (command === 'pm') {
-          //TODO parse the argument
-          message.channel.send(`send a private message: ${commandArgs}`);
-        }
-
-        if (command === 'gm') {
-          message.channel.send(`send a group message: ${commandArgs}`);
-        }
-
-        if (command === 'm') {
-          const usernames = await this.getMembers().map(m => m.username);
-          message.channel.send(`Here are all the members: ${usernames}`);
-        }
-      }
-    });
-
-     return client.login(token);
+ start() {
+    try {
+      const client = this.client;
+      const loginStatus = client.login(this.token);
+      client.once('ready', () => {
+        console.log('Discord Ready!');
+      });
+      return loginStatus;
+    } catch(err){
+      return err;
+    }
```

Comment:
we dont actually need try catch here, we could just `return the client.login(this.token)` promise and we could handle the error on user handle, basically we do want user know if an error is being thrown.
also if we really want to do the try catch, we could do the promise way of error handling sth like
`return client.login(this.token).catch(error=> error);` so it's more concise !

Reply:
fixed